citesphere/src/test/java/edu/asu/diging/citesphere/core/service/impl/ImportCrossrefJobManagerTest.java

```diff
@@ -0,0 +1,115 @@
+import edu.asu.diging.citesphere.model.bib.ICitationGroup;
+import edu.asu.diging.citesphere.user.IUser;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ImportCrossrefJobManagerTest {
+
+    @InjectMocks
+    private ImportCrossrefJobManager importCrossrefJobManager;
+
+    @Mock
+    private IGroupManager groupManager;
+
+    @Mock
+    private ImportCrossrefJobRepository jobRepo;
+
+    @Mock
+    private IKafkaRequestProducer kafkaProducer;
+
+    @Mock
+    private IJwtTokenService tokenService;
+    
+    @Mock
+    private IUser user;
+
+    @Mock
+    private ICitationGroup group;
+    private List<String> dois;
+
+    @Before
+    public void setup() {
+        ReflectionTestUtils.setField(importCrossrefJobManager, "jobPageSize", 10);
+        doReturn("username").when(user).getUsername();
+        dois = Arrays.asList("10.1000/xyz123", "10.1000/xyz456");
+    }
+
+    @Test
+    public void test_createJob_success() throws Exception {
+        String groupId = "groupId";
+        List<String> dois = Arrays.asList("doi1", "doi2");
+        doReturn(group).when(groupManager).getGroup(user, groupId);
+        doReturn("jobToken").when(tokenService).generateJobApiToken(any(ImportCrossrefJob.class));
+
+        IImportCrossrefJob job = importCrossrefJobManager.createJob(user, groupId, dois);
+
+        assertNotNull(job);
+        assertEquals(user.getUsername(), job.getUsername());
+        assertEquals(dois, job.getDois());
+        assertEquals(groupId, job.getCitationGroup());
+        assertEquals(JobStatus.PREPARED, job.getStatus());
+        verify(jobRepo, times(1)).save(any(ImportCrossrefJob.class));
+        verify(kafkaProducer).sendRequest(any(KafkaJobMessage.class), anyString());
+    }
+
+    @Test
+    public void test_createJob_groupDoesNotExist() {
+        String groupId = "groupId";
+        List<String> dois = Arrays.asList("doi1", "doi2");
+        doReturn(null).when(groupManager).getGroup(user, groupId);
+
+        GroupDoesNotExistException exception = assertThrows(GroupDoesNotExistException.class, () -> importCrossrefJobManager.createJob(user, groupId, dois));
+        assertNotNull(exception);
+    }
+
+    @Test
+    public void test_createJob_messageCreationException() throws Exception {
+        String groupId = "groupId";
+        when(groupManager.getGroup(user, groupId)).thenReturn(group);
+        when(tokenService.generateJobApiToken(any(ImportCrossrefJob.class))).thenReturn("token");
+        doThrow(new MessageCreationException("Error creating message")).when(kafkaProducer).sendRequest(any(KafkaJobMessage.class), anyString());
+
+        IImportCrossrefJob job = importCrossrefJobManager.createJob(user, groupId, dois);
+
+        assertNotNull(job);
+        assertEquals(JobStatus.FAILURE, job.getStatus());
+        assertEquals(1, job.getPhases().size());
+        assertEquals("Error creating message", job.getPhases().get(0).getMessage());
```

Comment:
this is not worth checking; if the text changes the method logic won't change but it'll make the test fail.